lgc/test/scalarizationOfDescriptorLoadsTest1.ll

```diff
@@ -0,0 +1,113 @@
+; NOTE: Assertions have been autogenerated by tool/update_llpc_test_checks.py UTC_ARGS: --tool lgc
+; RUN: lgc -mcpu=gfx1010 -print-after=lgc-builder-replayer -o - %s 2>&1 | FileCheck --check-prefixes=CHECK %s
+; ModuleID = 'lgcPipeline'
+source_filename = "lgcPipeline"
+target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-p7:160:256:256:32-p8:128:128-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-S32-A5-G1-ni:7:8-p32:32:32"
+target triple = "amdgcn--amdpal"
+
+; Function Attrs: nounwind
+define dllexport spir_func void @lgc.shader.VS.main() local_unnamed_addr #0 !spirv.ExecutionModel !14 !lgc.shaderstage !15 {
+.entry:
+  %0 = call <4 x i32> (...) @lgc.create.read.generic.input.v4i32(i32 2, i32 0, i32 0, i32 0, i32 0, i32 poison)
+  %.fr = freeze <4 x i32> %0
+  %__llpc_input_proxy_4.0.vec.extract = extractelement <4 x i32> %.fr, i64 0
+  %__llpc_input_proxy_4.4.vec.extract = extractelement <4 x i32> %.fr, i64 1
+  %1 = call ptr addrspace(4) (...) @lgc.create.get.desc.ptr.p4(i32 4, i32 4, i64 0, i32 0)
+  %2 = call i32 (...) @lgc.create.get.desc.stride.i32(i32 4, i32 4, i64 0, i32 0)
+  %3 = mul i32 %__llpc_input_proxy_4.4.vec.extract, %2
+  %4 = sext i32 %3 to i64
+  %5 = getelementptr i8, ptr addrspace(4) %1, i64 %4
+  %6 = load <4 x i32>, ptr addrspace(4) %5, align 16, !invariant.load !16
+  %7 = mul i32 %__llpc_input_proxy_4.0.vec.extract, %2
+  %8 = sext i32 %7 to i64
+  %9 = getelementptr i8, ptr addrspace(4) %1, i64 %8
+  %10 = load <4 x i32>, ptr addrspace(4) %9, align 16, !invariant.load !16
+  %11 = call reassoc nnan nsz arcp contract afn <4 x float> (...) @lgc.create.image.load.v4f32(i32 0, i32 8, <4 x i32> %10, i32 0)
+  call void (...) @lgc.create.image.store(<4 x float> %11, i32 0, i32 8, <4 x i32> %6, i32 1)
+  ret void
+}
+
+; Function Attrs: nounwind willreturn memory(read)
+declare <4 x i32> @lgc.create.read.generic.input.v4i32(...) local_unnamed_addr #1
+
+; Function Attrs: nounwind memory(none)
+declare ptr addrspace(4) @lgc.create.get.desc.ptr.p4(...) local_unnamed_addr #2
+
+; Function Attrs: nounwind memory(none)
+declare i32 @lgc.create.get.desc.stride.i32(...) local_unnamed_addr #2
+
+; Function Attrs: nounwind willreturn memory(read)
+declare <4 x float> @lgc.create.image.load.v4f32(...) local_unnamed_addr #1
+
+; Function Attrs: nounwind memory(write)
+declare void @lgc.create.image.store(...) local_unnamed_addr #3
+
+attributes #0 = { nounwind "denormal-fp-math-f32"="preserve-sign" }
+attributes #1 = { nounwind willreturn memory(read) }
+attributes #2 = { nounwind memory(none) }
+attributes #3 = { nounwind memory(write) }
+
+!lgc.client = !{!0}
```

Comment:
Nit: can you remove the metadata? I think most of them are irrelevant.

Reply:
If I remove them, then the code does not run through llpc pipeline.